### PR TITLE
PHP 8.2 compatibility + phpstan fixes

### PR DESCRIPTION
**Pull request by iansltx**
Since this is a big PR (+1889,-2089 on PHP files) that I expect we'll have multiple folks reviewing, here's a list of all of its PHP files so we can make sure everything gets reviewed by at least one person:

## Misc

- [x] .php-cs-fixer.php
- [x] inc/nginx-helper.php
- [x] public/index.php
- [x] scripts/newConsumerKeyFunc.php
- [x] src/ContainerFactory.php
- [x] tools/dbgen/generate.php

## Controllers

- [x] src/Controller/ApplicationsController.php
- [x] src/Controller/BaseApiController.php
- [x] src/Controller/BaseTalkController.php
- [x] src/Controller/ContactController.php
- [x] src/Controller/DefaultController.php
- [x] src/Controller/EmailsController.php
- [x] src/Controller/EventCommentsController.php
- [x] src/Controller/EventHostsController.php
- [x] src/Controller/EventImagesController.php
- [x] src/Controller/EventsController.php
- [x] src/Controller/FacebookController.php
- [x] src/Controller/LanguagesController.php
- [x] src/Controller/TalkCommentsController.php
- [x] src/Controller/TalkLinkController.php
- [x] src/Controller/TalkTypesController.php
- [x] src/Controller/TalksController.php
- [x] src/Controller/TokenController.php
- [x] src/Controller/TracksController.php
- [x] src/Controller/TwitterController.php
- [x] src/Controller/UsersController.php

## Exceptions + Header

- [x] src/Exception/AuthenticationException.php
- [x] src/Exception/AuthorizationException.php
- [x] src/Exception/Comment/TalkCommentDeletionException.php
- [x] src/Header.php

## Models

- [x] src/Model/ApiMapper.php
- [x] src/Model/BaseModel.php
- [x] src/Model/BaseModelCollection.php
- [x] src/Model/ClientMapper.php
- [x] src/Model/ClientModel.php
- [x] src/Model/ClientModelCollection.php
- [x] src/Model/EventCommentMapper.php
- [x] src/Model/EventCommentReportModel.php
- [x] src/Model/EventCommentReportModelCollection.php
- [x] src/Model/EventHostMapper.php
- [x] src/Model/EventMapper.php
- [x] src/Model/LanguageMapper.php
- [x] src/Model/OAuthModel.php
- [x] src/Model/PendingTalkClaimMapper.php
- [x] src/Model/PendingTalkClaimModel.php
- [x] src/Model/PendingTalkClaimModelCollection.php
- [x] src/Model/TalkCommentMapper.php
- [x] src/Model/TalkCommentReportModel.php
- [x] src/Model/TalkCommentReportModelCollection.php
- [x] src/Model/TalkMapper.php
- [x] src/Model/TalkModel.php
- [x] src/Model/TalkModelCollection.php
- [x] src/Model/TalkTypeMapper.php
- [x] src/Model/TokenMapper.php
- [x] src/Model/TokenModel.php
- [x] src/Model/TokenModelCollection.php
- [x] src/Model/TrackMapper.php
- [x] src/Model/TwitterRequestTokenMapper.php
- [x] src/Model/TwitterRequestTokenModel.php
- [x] src/Model/TwitterRequestTokenModelCollection.php
- [x] src/Model/UserMapper.php

## Request + Router

- [x] src/Request.php
- [x] src/Router/ApiRouter.php
- [x] src/Router/BaseRouter.php
- [x] src/Router/DefaultRouter.php
- [x] src/Router/Route.php
- [x] src/Router/VersionedRouter.php

## Services

- [x] src/Service/BaseEmailService.php
- [x] src/Service/CommentReportedEmailService.php
- [x] src/Service/ContactEmailService.php
- [x] src/Service/EventApprovedEmailService.php
- [x] src/Service/EventCommentReportedEmailService.php
- [x] src/Service/EventRejectedEmailService.php
- [x] src/Service/EventSubmissionEmailService.php
- [x] src/Service/NullSpamCheckService.php
- [x] src/Service/SpamCheckService.php
- [x] src/Service/SpamCheckServiceInterface.php
- [x] src/Service/TalkAssignEmailService.php
- [x] src/Service/TalkClaimApprovedEmailService.php
- [x] src/Service/TalkClaimEmailService.php
- [x] src/Service/TalkClaimRejectedEmailService.php
- [x] src/Service/TalkCommentEmailService.php
- [x] src/Service/UserPasswordResetEmailService.php
- [x] src/Service/UserRegistrationEmailService.php
- [x] src/Service/UserUsernameReminderEmailService.php

## View

- [x] src/View/ApiView.php
- [x] src/View/HtmlView.php
- [x] src/View/JsonPView.php
- [x] src/View/JsonView.php

## Tests for Controllers

- [x] tests/ContainerFactoryTest.php
- [x] tests/Controller/ContactControllerTest.php
- [x] tests/Controller/EventHostsControllerTest.php
- [x] tests/Controller/TalkBase.php
- [x] tests/Controller/TalkLinkControllerTest.php
- [x] tests/Controller/TalkTypesControllerTest.php
- [x] tests/Controller/TalksControllerDeleteTest.php
- [x] tests/Controller/TalksControllerTest.php
- [x] tests/Controller/TokenControllerTest.php
- [x] tests/Controller/UsersControllerTest.php

## Tests for headers, model, request

- [x] tests/HeaderTest.php
- [x] tests/Model/ApiMapperTest.php
- [x] tests/Model/EventHostMapperTest.php
- [x] tests/Model/OauthModelTest.php
- [x] tests/Model/TalkMapperTest.php
- [x] tests/Model/TestApiMapper.php
- [x] tests/RequestTest.php

## Tests for Router

- [x] tests/Router/ApiRouterTest.php
- [x] tests/Router/DefaultRouterTest.php
- [x] tests/Router/RouteTest.php
- [x] tests/Router/RouterTest.php
- [x] tests/Router/TestController3.php
- [x] tests/Router/TestRouter3.php
- [x] tests/Router/VersionedRouterTest.php

## Tests for services, view

- [x] tests/Service/CommentReportedEmailServiceTest.php
- [x] tests/Service/NullSpamCheckServiceTest.php
- [x] tests/Service/SpamCheckServiceTest.php
- [x] tests/Service/TalkCommentEmailServiceTest.php
- [x] tests/View/ApiViewTest.php
- [x] tests/View/JsonViewTest.php

### CODE REVIEW

**Comment by iansltx** [comment]
Figure we can just solve the Stan errors while we're at this, since there are "only" 676 of them across 90 files.

**Review by heiglandreas — DISMISSED** [review]
Amazing work! Looks good to me, so I definitely approve that. 

There are some comments that should be addressed in one or more future PRs but that would not stop me from merging this!

Only in the `EventCommentMapper` there seems to be something missing! Before merging someone should have a look at that.

File: src/Controller/LanguagesController.php
Comment (Synchro) [reply src/Controller/LanguagesController.php:13]: This was the result of 2 successive changes I did. Originally the function didn't have a return type, and `getLanguageById` could return false, so I set the return type to match that. Later I added an explicit check for `false`, so it will no longer be returned, but didn't update the return type to match.

File: src/Controller/TalkTypesController.php
Comment (Synchro) [reply src/Controller/TalkTypesController.php:27]: Same reason as [this](https://github.com/joindin/joindin-api/pull/944#discussion_r1219088148)

File: src/Model/EventCommentMapper.php
Comment (herndlm) [reply src/Model/EventCommentMapper.php:331]: hey, since I messed this up, feel free to ping me :) I'll fix it. this happens if you only look blindly at phpstan errors and mix some "tiny" refactoring in...

Comment (iansltx) [reply src/Model/EventCommentMapper.php:331]: @herndlm PR in the fix and I'll happily take it :). I would've either fixed or reached out but, well, I'm still awake in UTC-5 for work reasons 🙃

**Comment by iansltx** [comment]
Merging this now. Will deploy over lunch in a few hours...and then fix whatever breaks.

**Pull request merged**
